### ISSUE TITLE
Camel case declared properties with single letter in the middle are being ignored

### ISSUE THREAD

**Issue by alexclemensua** [Polymer/polymer#3206]
Camel case declared properties with single letter in the middle are being ignored. This behaviour has been introduced somewhere around 1.2.3, because in 1.0.0, 1.1.0 and 1.1.3 it works as expected. If this is not considered a bug then information about the new behaviour should be included in the [documentation](https://www.polymer-project.org/1.0/docs/devguide/properties.html#attribute-deserialization).

For example: a boolean variable named alignYAxis will be ignored when added to the element code:

    <chart-nasty align-y-axis>a very informative chart goes here</chart-nasty>

    Polymer({
      is : 'chart-nasty',
      properties: {
        alignYAxis: Boolean
      }
    });

but alignAxis works as expected:

    <chart-fine align-axis>an even more informative chart goes here</chart-fine>

    Polymer({
      is : 'chart-fine',
      properties: {
        alignAxis: Boolean
      }
    });

**Comment by arthurevans** [comment]
Looks like this change was introduced in 1.2.2 as a result of the mirco-optimizations, I think specifically of this change here: https://github.com/Polymer/polymer/commit/acdd242616f77616f1f3f90e0389c4c98161bcb4

We used to do dash-case to camel-case when processing the attributes, but now we build up a `_propertyInfo` object using the `camelCaseToDashCase` function, which doesn't handle consecutive capital letters.

I'd definitely argue it's a bug, since it's a) a change in behavior, and b) the `camelCaseToDashCase` function doesn't correctly reverse the `dashCaseToCamelCase` function in `case-map.html`.

**Comment by TimvdLippe** [comment]
@arthurevans Looking at the git history, it appears that the `camelCaseToDashCase` function never worked with 2 uppercase characters in sequence. Is that correct?

[Renaming of file](https://github.com/Polymer/polymer/commit/06f78ee87e8db48bfb5816550960fab6af1b2e88)
[Caching of conversions](https://github.com/Polymer/polymer/commit/ee3e0fda4b735726978168dca8939252f7af3811)
[Moving the function again](https://github.com/Polymer/polymer/commit/6c9e396a1bc9f14fa3dbd3dd7cb06a05737d8832)
[First version of the function](https://github.com/Polymer/polymer/commit/a9d714a3294414b49e978634db6ad7577428c134)

**Comment by jimsimon** [comment]
This doesn't seem like a bug per the docs:
> When mapping attribute names to property names:
>
> Attribute names are converted to lowercase property names. For example, the attribute firstName maps to firstname.
> 
> Attribute names with dashes are converted to camelCase property names by capitalizing the character following each dash, then removing the dashes. For example, the attribute first-name maps to firstName.

In this case an attribute name of ```alignYAxis``` would have a corresponding property name of ```alignyaxis``` due to the first rule:
```
<chart alignYAxis>a very informative chart goes here</chart>

Polymer({
  is : 'chart'
  properties: {
    alignyaxis: Boolean
  }
});
```

If you truly want a camel cased property name, you'd need to take advantage of the second rule by changing your attribute name to ```align-y-axis```:
```
<chart align-y-axis>a very informative chart goes here</chart>

Polymer({
  is : 'chart'
  properties: {
    alignYAxis: Boolean
  }
});
```

**Comment by alexclemensua** [comment]
@jimsimon you are right. my example was flawed, I've edited my first post to make it precise. I also made this plnkr for illustration purposes: http://plnkr.co/TmCQ8WS9d19jULFohSaE

**Comment by TimvdLippe** [comment]
Fixed in #3279
